types/src/transaction/webauthn.rs

```diff
@@ -118,6 +126,11 @@ impl PartialAuthenticatorAssertionResponse {
         message: &T,
         public_key: &AnyPublicKey,
     ) -> Result<()> {
+        // PartialAuthenticatorAssertionResponse should not exceed MAX_WEBAUTHN_SIGNATURE_BYTES
```

Comment:
@gedigi but we are reserializing the signature here, which is inefficient...

This check should be done at the deserialization time instead...

Reply:
performing this check at deserialization works too and is definitely more efficient. though the goal of the change was to group all checks in one place to avoid having to track that all checks are done correctly and at the right time.

Reply:
It will definitely happen on validators too though. They most definitely run `verify`.

In any case, we're unnecessarily serializing this, which is odd... 

I agree that the code is harder to track if we put it in the right place (i.e., when deserialization is done).

I just don't like wasting performance here.

Reply:
the new code performs the check without serializing

Reply:
I still think this is the wrong way to do it because it's not modular.

e.g., in keyless signatures, I have an ephemeral signature which could be of type `PartialAuthenticatorAssertionResponse` (damn it @hariria; that name... :P)

```rust
pub enum EphemeralSignature {
    Ed25519 {
        signature: Ed25519Signature,
    },
    WebAuthn {
        signature: PartialAuthenticatorAssertionResponse,
    },
}
```

I now have to replicate this max size logic in my keyless code, when enforcing a max size for its ephemeral signature. This is not good, since one might forget to do this, or one might forget to update the code in all places.

So doing this at actual deserialization time is preferred...

Reply:
Ah, nevermind, my keyless signature code will call this code. 🤦

Reply:
I agree with @gedigi and I think this makes more sense than adding an arbitrary size limit on the struct itself.

Reply:
@alinush any updates on this?

Reply:
I think we should rely on the TXN limits until we can code our own limits properly.